[Bug] pnpm install fails with "No matching version found for @storybook/addon-styling-webpack@^7.3.2"

### Describe the bug
Trying to upgrade and just do just this:
```
# pnpm dlx storybook@latest add @storybook/addon-styling-webpack                                                               
.../share/pnpm/store/v3/tmp/dlx-16529    | +616 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
.../share/pnpm/store/v3/tmp/dlx-16529    | Progress: resolved 637, reused 616, downloaded 0, added 616, done
Verifying @storybook/addon-styling-webpack
Installing @storybook/addon-styling-webpack@^7.3.2

An error occurred while installing dependencies:
PNPM error ERR_PNPM_NO_MATCHING_VERSION  No matching version found for @storybook/addon-styling-webpack@^7.3.2

Please check the logfile generated at ./storybook.log for troubleshooting and try again.
/Users/ar/.local/share/pnpm/store/v3/tmp/dlx-16529/node_modules/.pnpm/@storybook+cli@7.3.2/node_modules/@storybook/cli/dist/generate.js:54
An error occurred while installing dependencies:`),logger4.log(e.message),new HandledError(e)}}async removeDependencies(options,dependencies){let{skipInstall}=options;if(skipInstall){let{packageJson}=options;dependencies.forEach(dep=>{packageJson.devDependencies&&delete packageJson.devDependencies[dep],packageJson.dependencies&&delete packageJson.dependencies[dep]}),await this.writePackageJson(packageJson)}else try{await this.runRemoveDeps(dependencies)}catch(e){throw logger4.error("An error occurred while removing dependencies."),logger4.log(e.message),new HandledError(e)}}getVersionedPackages(packages){return Promise.all(packages.map(async pkg2=>{let[packageName,packageVersion]=getPackageDetails(pkg2);return`${packageName}@${await this.getVersion(packageName,packageVersion)}`}))}getVersions(...packageNames){return Promise.all(packageNames.map(packageName=>this.getVersion(packageName)))}async getVersion(packageName,constraint){let current;/(@storybook|^sb$|^storybook$)/.test(packageName)&&(current=versions_default[packageName]);let latest;try{latest=await this.latestVersion(packageName,constraint)}catch(e){if(current)return logger4.warn(`
                                                                          ^

HandledError: PNPM error ERR_PNPM_NO_MATCHING_VERSION  No matching version found for @storybook/addon-styling-webpack@^7.3.2

Please check the logfile generated at ./storybook.log for troubleshooting and try again.
    at PNPMProxy.addDependencies (/Users/ar/.local/share/pnpm/store/v3/tmp/dlx-16529/node_modules/.pnpm/@storybook+cli@7.3.2/node_modules/@storybook/cli/dist/generate.js:54:75)
    at async add (/Users/ar/.local/share/pnpm/store/v3/tmp/dlx-16529/node_modules/.pnpm/@storybook+cli@7.3.2/node_modules/@storybook/cli/dist/generate.js:127:1955) {
  handled: true,
  cause: Error: PNPM error ERR_PNPM_NO_MATCHING_VERSION  No matching version found for @storybook/addon-styling-webpack@^7.3.2

  Please check the logfile generated at ./storybook.log for troubleshooting and try again.
      at PNPMProxy.runAddDeps (/Users/ar/.local/share/pnpm/store/v3/tmp/dlx-16529/node_modules/.pnpm/@storybook+cli@7.3.2/node_modules/@storybook/cli/dist/generate.js:58:5082)
      at async PNPMProxy.addDependencies (/Users/ar/.local/share/pnpm/store/v3/tmp/dlx-16529/node_modules/.pnpm/@storybook+cli@7.3.2/node_modules/@storybook/cli/dist/generate.js:53:853)
      at async add (/Users/ar/.local/share/pnpm/store/v3/tmp/dlx-16529/node_modules/.pnpm/@storybook+cli@7.3.2/node_modules/@storybook/cli/dist/generate.js:127:1955)
}

Node.js v20.5.1
```

the logfile mentioned looks like
```
Progress: resolved 0, reused 1, downloaded 0, added 0
Progress: resolved 86, reused 86, downloaded 0, added 0
 ERR_PNPM_NO_MATCHING_VERSION  No matching version found for @storybook/addon-styling-webpack@^7.3.2

This error happened while installing a direct dependency of /Users/ar/src/lincare/classifier/classifier-gui

The latest release of @storybook/addon-styling-webpack is "0.0.2".

If you need the full list of all 2 published versions run "$ pnpm view @storybook/addon-styling-webpack versions".
```

### Steps to reproduce the behavior

1. be stupid and run pnpm dlx ...

### Expected behavior

A clear and concise description of what you expected to happen.

### Screenshots and/or logs

If applicable, add screenshots and/or logs to help explain your problem.

### Environment

 - OS: Mac OSX 13.4.1
 - Node.js version: v20.5.1
 - pNPM version: 8.6.12

### Additional context

i'm stupid...

Just tried to install this on top of an existing 7.0 installation...

--- a/code/lib/cli/src/add.ts
+++ b/code/lib/cli/src/add.ts
@@ -7,7 +7,7 @@ import {
   useNpmWarning,
   type PackageManagerName,
 } from './js-package-manager';
-import { getStorybookVersion } from './utils';
+import { getStorybookVersion, isCorePackage } from './utils';
 
 const logger = console;
 
@@ -76,8 +76,9 @@ export async function add(
 
   // add to package.json
   const isStorybookAddon = addonName.startsWith('@storybook/');
+  const isAddonFromCore = isCorePackage(addonName);
   const storybookVersion = await getStorybookVersion(packageManager);
-  const version = versionSpecifier || (isStorybookAddon ? storybookVersion : latestVersion);
+  const version = versionSpecifier || (isAddonFromCore ? storybookVersion : latestVersion);
   const addonWithVersion = SemVer.valid(version)
     ? `${addonName}@^${version}`
     : `${addonName}@${version}`;

--- a/code/lib/cli/src/upgrade.test.ts
+++ b/code/lib/cli/src/upgrade.test.ts
@@ -1,9 +1,4 @@
-import {
-  addExtraFlags,
-  addNxPackagesToReject,
-  getStorybookVersion,
-  isCorePackage,
-} from './upgrade';
+import { addExtraFlags, addNxPackagesToReject, getStorybookVersion } from './upgrade';
 
 describe.each([
   ['│ │ │ ├── @babel/code-frame@7.10.3 deduped', null],
@@ -23,24 +18,6 @@ describe.each([
 ])('getStorybookVersion', (input, output) => {
   it(`${input}`, () => {
     expect(getStorybookVersion(input)).toEqual(output);
-  });
-});
-
-describe.each([
-  ['@storybook/react', true],
-  ['@storybook/node-logger', true],
-  ['@storybook/addon-info', true],
-  ['@storybook/something-random', true],
-  ['@storybook/preset-create-react-app', false],
-  ['@storybook/linter-config', false],
-  ['@storybook/design-system', false],
-  ['@storybook/addon-styling', false],
-  ['@storybook/addon-styling-webpack', false],
-  ['@nx/storybook', false],
-  ['@nrwl/storybook', false],
-])('isCorePackage', (input, output) => {
-  it(`${input}`, () => {
-    expect(isCorePackage(input)).toEqual(output);
   });
 });
 

--- a/code/lib/cli/src/upgrade.ts
+++ b/code/lib/cli/src/upgrade.ts
@@ -8,6 +8,7 @@ import type { PackageJsonWithMaybeDeps, PackageManagerName } from './js-package-
 import { getPackageDetails, JsPackageManagerFactory, useNpmWarning } from './js-package-manager';
 import { commandLog } from './helpers';
 import { automigrate } from './automigrate';
+import { isCorePackage } from './utils';
 
 type Package = {
   package: string;
@@ -24,37 +25,6 @@ export const getStorybookVersion = (line: string) => {
     version: match[2],
   };
 };
-
-const excludeList = [
-  '@storybook/addon-bench',
-  '@storybook/addon-console',
-  '@storybook/addon-postcss',
-  '@storybook/addon-styling',
-  '@storybook/addon-styling-webpack',
-  '@storybook/babel-plugin-require-context-hook',
-  '@storybook/bench',
-  '@storybook/builder-vite',
-  '@storybook/csf',
-  '@storybook/design-system',
-  '@storybook/ember-cli-storybook',
-  '@storybook/eslint-config-storybook',
-  '@storybook/expect',
-  '@storybook/jest',
-  '@storybook/linter-config',
-  '@storybook/mdx1-csf',
-  '@storybook/mdx2-csf',
-  '@storybook/react-docgen-typescript-plugin',
-  '@storybook/storybook-deployer',
-  '@storybook/test-runner',
-  '@storybook/testing-library',
-  '@storybook/testing-react',
-  '@nrwl/storybook',
-  '@nx/storybook',
-];
-export const isCorePackage = (pkg: string) =>
-  pkg.startsWith('@storybook/') &&
-  !pkg.startsWith('@storybook/preset-') &&
-  !excludeList.includes(pkg);
 
 const deprecatedPackages = [
   {

--- a/code/lib/cli/src/utils.test.ts
+++ b/code/lib/cli/src/utils.test.ts
@@ -1,0 +1,21 @@
+import { isCorePackage } from './utils';
+
+describe('UTILS', () => {
+  describe.each([
+    ['@storybook/react', true],
+    ['@storybook/node-logger', true],
+    ['@storybook/addon-info', true],
+    ['@storybook/something-random', true],
+    ['@storybook/preset-create-react-app', false],
+    ['@storybook/linter-config', false],
+    ['@storybook/design-system', false],
+    ['@storybook/addon-styling', false],
+    ['@storybook/addon-styling-webpack', false],
+    ['@nx/storybook', false],
+    ['@nrwl/storybook', false],
+  ])('isCorePackage', (input, output) => {
+    it(`It should return "${output}" when given "${input}"`, () => {
+      expect(isCorePackage(input)).toEqual(output);
+    });
+  });
+});

--- a/code/lib/cli/src/utils.ts
+++ b/code/lib/cli/src/utils.ts
@@ -97,3 +97,34 @@ export const createLogStream = async (
     logStream.once('error', reject);
   });
 };
+
+const PACKAGES_EXCLUDED_FROM_CORE = [
+  '@storybook/addon-bench',
+  '@storybook/addon-console',
+  '@storybook/addon-postcss',
+  '@storybook/addon-styling',
+  '@storybook/addon-styling-webpack',
+  '@storybook/babel-plugin-require-context-hook',
+  '@storybook/bench',
+  '@storybook/builder-vite',
+  '@storybook/csf',
+  '@storybook/design-system',
+  '@storybook/ember-cli-storybook',
+  '@storybook/eslint-config-storybook',
+  '@storybook/expect',
+  '@storybook/jest',
+  '@storybook/linter-config',
+  '@storybook/mdx1-csf',
+  '@storybook/mdx2-csf',
+  '@storybook/react-docgen-typescript-plugin',
+  '@storybook/storybook-deployer',
+  '@storybook/test-runner',
+  '@storybook/testing-library',
+  '@storybook/testing-react',
+  '@nrwl/storybook',
+  '@nx/storybook',
+];
+export const isCorePackage = (pkg: string) =>
+  pkg.startsWith('@storybook/') &&
+  !pkg.startsWith('@storybook/preset-') &&
+  !PACKAGES_EXCLUDED_FROM_CORE.includes(pkg);

